### PR TITLE
Refactor css classes into single file for table

### LINKED ISSUE #438: Lift out table class names into common file
All of the components in the table package currently have their unique identifying class name hardcoded in JSX. We should consider lifting these class names into a common file such as a new `Classes.ts` to make it easier to share and audit class names used in the package.

### PR DESCRIPTION
**Pull request by gscshoyru**
Same as the other packages.

#### Reviewers should focus on:

Please check my work! This is straightforward, but there was a lot of it.

### CODE REVIEW

File: packages/table/src/common/classes.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
```
Comment (llorca) [inline packages/table/src/common/classes.ts]: 2017 I believe (creation year)

File: packages/table/src/cell/cell.tsx [hunk 2]
```diff
@@ -8,8 +8,9 @@
 import * as classNames from "classnames";
 import * as PureRender from "pure-render-decorator";
 import * as React from "react";
+import * as Classes from "../common/classes";
 
-import { Classes, IIntentProps, IProps } from "@blueprintjs/core";
+import { Classes as BlueprintClasses, IIntentProps, IProps } from "@blueprintjs/core";
```
Comment (llorca) [inline packages/table/src/cell/cell.tsx]: `BlueprintClasses` vs `CoreClasses`?

File: packages/table/src/cell/formats/truncatedFormat.tsx [hunk 3]
```diff
@@ -87,33 +90,39 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
...
             const popoverContent = <div className={popoverClasses}>{children}</div>;
-            const className = classNames(this.props.className, "bp-table-truncated-format");
+            const className = classNames(this.props.className, Classes.TABLE_TRUNCATED_FORMAT);
             const constraints = [{
                 attachment: "together",
                 pin: true,
                 to: "window",
             }];
+
+            const iconClasses = classNames (
```
Comment (giladgray) [inline packages/table/src/cell/formats/truncatedFormat.tsx]: no space after function name

File: packages/table/src/common/classes.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+export const TABLE_BODY = "bp-table-body";
+export const TABLE_NO_HORIZONTAL_SCROLL = "bp-table-no-horizontal-scroll";
+export const TABLE_NO_VERTICAL_SCROLL = "bp-table-no-vertical-scroll";
+export const TABLE_BODY_SCROLL_CLIENT = "bp-table-body-scroll-client";
+export const TABLE_RESIZE_GUIDES = "bp-table-resize-guides";
+export const TABLE_BODY_VIRTUAL_CLIENT = "bp-table-body-virtual-client";
+export const TABLE_CELL_CLIENT = "bp-table-cell-client";
+export const TABLE_CELL_GHOST = "bp-table-cell-ghost";
+export const TABLE_CELL_LEDGER_ODD = "bp-table-cell-ledger-odd";
+export const TABLE_CELL_LEDGER_EVEN = "bp-table-cell-ledger-even";
```
Comment (giladgray) [inline packages/table/src/common/classes.ts]: wow please organize these _somehow_. at the very least alphabetize. at the most, sort them by usefulness, then by alph.

File: packages/table/src/cell/cell.tsx (hunk 2 above)
Comment (gscshoyru) [inline packages/table/src/cell/cell.tsx]: I could have sworn I was following an example I found somewhere else, but I can't seem to find it. If this is preferred I can change it.

**Comment by gscshoyru** [comment]
@giladgray @llorca anything else needed here?

**Comment by llorca** [comment]
LGTM, we're not shipping this week cause other folks are out hacking in NY. Suggest waiting next week for proper CR

File: packages/table/src/common/classes.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+export const TABLE_SELECTION_REGION = "bp-table-selection-region";
+export const TABLE_TH_MENU = "bp-table-th-menu";
+export const TABLE_THEAD = "bp-table-thead";
+export const TABLE_TOP_CONTAINER = "bp-table-top-container";
+export const TABLE_TRUNCATED_FORMAT = "bp-table-truncated-format";
+export const TABLE_TRUNCATED_POPOVER = "bp-table-truncated-popover";
+export const TABLE_TRUNCATED_POPOVER_TARGET = "bp-table-truncated-popover-target";
+export const TABLE_TRUNCATED_TEXT = "bp-table-truncated-text";
+export const TABLE_TRUNCATED_VALUE = "bp-table-truncated-value";
+export const TABLE_VERTICAL_GUIDE = "bp-table-vertical-guide";
```
Comment (themadcreator) [inline packages/table/src/common/classes.ts]: Man, i didn't realize how many dang classes we have until seeing this whole list

File: packages/table/src/common/classes.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+/** Common code for row and column index class generator functions, since they're essentially the same */
```
Comment (themadcreator) [inline packages/table/src/common/classes.ts]: I think standard typedoc format is to have a newline after the `/**` like so:

```
/**
 * Common code for...
 */
```

Not sure if it'd change how it's rendered

File: packages/table/src/tableBody.tsx [hunk 7]
```diff
@@ -146,14 +142,20 @@ export class TableBody extends React.Component<ITableBodyProps, {}> {
...
+
+        let currentSelectedReigons = selectedRegions;
 
         if (renderBodyContextMenu == null) {
             return undefined;
         }
 
         const target = this.locateClick(e.nativeEvent as MouseEvent);
-        return renderBodyContextMenu(new MenuContext(target, selectedRegions, grid.numRows, grid.numCols));
+        if (!Regions.containsRegion(selectedRegions, target)) {
```
Comment (themadcreator) [inline packages/table/src/tableBody.tsx]: is this a bugfix?

File: packages/table/src/common/classes.ts (hunk 6 above)
Comment (giladgray) [inline packages/table/src/common/classes.ts]: this single-line format is perfectly acceptable, though this line is debatably long enough to wrap. 
rendering is unaffected by lines.

Comment (giladgray) [inline packages/table/src/common/classes.ts]: @gscshoyru our convention is to use periods at the end of documentation lines.

File: packages/table/src/common/classes.ts (hunk 5 above)
Comment (gscshoyru) [inline packages/table/src/common/classes.ts]: I know, right?

File: packages/table/src/tableBody.tsx (hunk 7 above)
Comment (gscshoyru) [inline packages/table/src/tableBody.tsx]: Uh. Whoops. This should not be in here. Thanks for catching.

**Pull request merged**